### PR TITLE
cluster park map markers

### LINKED ISSUE #35: Cluster markers
I think we can do this by extending a react-leaflet component to call addMarker/removeMarker from https://github.com/Leaflet/Leaflet.markercluster.

Related to #26

### PR DESCRIPTION
**Pull request by paulserraino**
closes #35 :dancer: 

lemme know what you guys think

### CODE REVIEW

**Comment by luqmaan** [comment]
Maybe you should call your package `react-leaflet-geojson-cluster` instead of `react-geojson-cluster`?

File: client/js/components/ParkMap.jsx [hunk 1]
```diff
@@ -53,6 +54,25 @@ function pointToLayer(feature, latlng) {
     return L.marker(latlng, {icon: iconLayer});
 }
 
+function clusterIcon (cluster) {
+    //majority feature wins
```
Comment (luqmaan) [inline client/js/components/ParkMap.jsx]: Space after the // please.

``` js
// majority feature wins
```

Comment (luqmaan) [inline client/js/components/ParkMap.jsx]: https://github.com/airbnb/javascript

**Comment by paulserraino** [comment]
:+1: I'm cool with that

File: client/scss/main.scss [hunk 2]
```diff
@@ -348,3 +348,10 @@ a {
     text-align: center;
     font-size: 30px;
 }
+
+.leaflet-cluster-anim .leaflet-marker-icon, .leaflet-cluster-anim .leaflet-marker-shadow {
+  -webkit-transition: -webkit-transform 0.3s ease-out, opacity 0.3s ease-in;
+  -moz-transition: -moz-transform 0.3s ease-out, opacity 0.3s ease-in;
+  -o-transition: -o-transform 0.3s ease-out, opacity 0.3s ease-in;
+  transition: transform 0.3s ease-out, opacity 0.3s ease-in;
+  }
```
Comment (luqmaan) [inline client/scss/main.scss]: Too much indent

File: client/js/components/ParkMap.jsx (hunk 1 above)
Comment (paulserraino) [inline client/js/components/ParkMap.jsx]: nice catch

**Comment by luqmaan** [comment]
The animations are beautiful.

Should we show this blue overlay? It seems a little odd since it only appears on hover (and most people will be using it the app on mobile).

image

Also, would it be useful to show the number of collapsed markers? Kind of like the cluster example plugin does?

**Comment by luqmaan** [comment]
Also, at the max zoom some icons are still clustered.

See http://open-austin.github.io/austingreenmap/#park/365

With clustering:

image

Without clustering:

image

**Comment by paulserraino** [comment]
> Should we show this blue overlay? It seems a little odd since it only appears on hover (and most people will be using it the app on mobile).

I think it would look better without the blue overlay. I haven't looked into yet, but it shouldn't be too difficult to remove it.

> Also, would it be useful to show the number of collapsed markers? Kind of like the cluster example plugin does?

I'm still unsure of this, because we have so many different types of park features, showing a number maybe too ambiguous.

> Also, at the max zoom some icons are still clustered.

I'll fix that, we should be able to adjust the zoom level

**Comment by luqmaan** [comment]
> > Also, would it be useful to show the number of collapsed markers? Kind of like the cluster example plugin does?
> 
> I'm still unsure of this, because we have so many different types of park features, showing a number maybe too ambiguous.

Yeah. :confused:  Lets just roll with most common icon wins for now.

**Comment by paulserraino** [comment]
I'm seeing incorrect popup fomat. Looking into a fix ...
screen shot 2015-07-27 at 9 01 20 pm

It should look like this:
screen shot 2015-07-27 at 9 03 39 pm

**Comment by luqmaan** [comment]
:deciduous_tree: @paulserraino feel free to merge whenever you're comfortable with it.

I added you as a contributor.

**Comment by luqmaan** [comment]
Haha, actually I'll merge this. Merging my park labels branch makes this conflict.

**Comment by paulserraino** [comment]
sounds good :+1:

**Comment by luqmaan** [comment]
Merged and deployed. See http://open-austin.github.io/austingreenmap/

**Comment by paulserraino** [comment]
awesome! :dancer:

**Comment by johntyree** [comment]
Nice

**Pull request closed without merging**